addons/account/models/account_move.py

```diff
@@ -2388,6 +2382,28 @@ def write(self, vals):
 
         return res
 
+    def _get_fields_used_by_hash(self):
+        fields_used = list(INTEGRITY_HASH_MOVE_FIELDS)
+        for line_field in INTEGRITY_HASH_LINE_FIELDS:
+            fields_used.append(f'line_ids.{line_field}')
+        return fields_used
```

Comment:
Maybe declare the fields directly with the prepended `line_ids` in `INTEGRITY_HASH_MOVE_FIELDS`? 
Not sure if that is used elsewhere, but if it is then it should be taken into account in the mixin anyways

Reply:
`INTEGRITY_HASH_MOVE_FIELDS` is used in `account_move`'s `write()`method whereas `INTEGRITY_HASH_LINE_FIELDS`is used in `account_move_line`'s `write()` method to check that no field used by the hash has been modified. Therefore we cannot use `line_ids.line_field` in `account_move_line` since this attribute would not exist when fetching it with `_get_fields_used_by_hash`

Reply:
These checks should not be done in `account.move`, but in `hash.mixin`

Reply:
Yes that is what has been done for `INTEGRITY_HASH_MOVE_FIELDS`.
However for `INTEGRITY_HASH_LINE_FIELDS`, it's a bit trickier since `account_move_line` does not inherit from `hash_mixin` (at least it shouldn't be the case I think ?)